fields/types/gcsavatar/GcsAvatarType.js

```diff
@@ -236,7 +236,18 @@ gcsavatar.prototype.uploadFile = function(item, file, update, callback) {
   const publicRead = _this.options.publicRead ? _this.options.publicRead : false;
   let filename = file.name;
   const split = filename.split('.');
-  const filenameWithoutExt = split[0];
+  const fileExt = split[1];
+
+  // Overwrite filename with username (i.e. email account prefix).
+  // e.g. When an user uploads the avatar with email 'alice@twreporter.org',
+  // the filename of the uploaded image would be 'alice.jpg'.
+  // The reason for doing this is that we don't need to deal with url 
+  // reset in the cookie for the keystone-plugin.
+  const email = _.get(item, '_doc.email');
+  const username = email.match(/^([^@]*)@/)[1];
+  const filenameWithoutExt = username;
+  filename = `${username}.${fileExt}`;
```

Comment:
Would we want to truncate the non-alphanumeric characters(e.g. `.`) which might result in multiple extension?

Reply:
Also, there might be cases that result in the filename duplication.
For instance,

alice@twreporter.org
alice@gmail.com

In such case, I would suggest adding the suffix => alice_twreporter, alice_gmail to avoid possible duplication.

Reply:
To address filename duplication issue, how about using email as the filename ?

Reply:
Yeah, that also works but you should also consider truncating the invalid characters.

Reply:
Updated.